contracts/zero-ex/contracts/src/storage/LibNativeOrdersStorage.sol

```diff
@@ -0,0 +1,54 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "./LibStorage.sol";
+
+
+/// @dev Storage helpers for `NativeOrdersFeature`.
+library LibNativeOrdersStorage {
+
+    /// @dev Storage bucket for this feature.
+    struct Storage {
+        // How much taker token has been filled in order.
+        // The lower `uint128` is the taker token fill amount.
+        // The high bit will be `1` if the order was directly cancelled.
+        mapping(bytes32 => uint256) orderHashToTakerTokenFilledAmount;
+        // The minimum valid order salt for a given maker and order pair (maker, taker)
+        // for limit orders.
+        mapping(address => mapping(address => mapping(address => uint256)))
```

Comment:
I think we can save ~60 gas per order if we flatten this mapping to be `keccak256(abi.encodePacked(maker, makerToken, takerToken)) => uint256`
probably not worth but just throwing it out there

Reply:
If you're checking on this, I'd try `abi.encode()` in addition to `abi.encodePacked()`. I don't have a good guess as to which is chaper.